lib/max_subarray.rb

```diff
@@ -1,8 +1,14 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def max_sub_array(nums)
```

Comment:
This works, but you can actually do it without the array.  You only need to keep track of the max subarray ending at the current index and the max subarray seen so far.  

It's kind of like the fibonacci problem.

That said your time complexity is great.